### PR TITLE
Simplified interface for invoking API Extractor

### PR DESCRIPTION
**Pull request by pgonzal**
This PR untangles API Extractor from **gulp-core-build-typescript**, so that it can now be easily invoked from lightweight build scripts such as **office-ui-fabric-react**:

- I added a simple command-line interface with most of the settings coming from a JSON config file (which eventually will have settings to relax some of the stricter rules)
- In the simplest configuration, the compiler settings are just read from tsconfig.json with reasonable defaults for everything else
- The command-line interface passes its configuration object to a very simple interface, which also serves as the public API for more advanced scripts that need to pass in a compiler state object
- All logic from **gulp-core-build-typescript** has been moved into **api-extractor** itself, avoiding the need for other build scripts to duplicate this logic
- The public API for **api-extractor** no longer exposes the internal syntax tree or tsc data structures; we are now one PR away from a fully cleaned up API (just need to remove some deprecated definitions)

Notes:
- A number of unit tests have been temporarily disabled, because they were invoking the old API directly.  I will reenable them when we switch over to Jest.
- In a separate PR I will rename Extractor->ExtractorContext and ApiExtractor->Extractor; it's not in this PR because it would make the diff hard to read

### CODE REVIEW

**Comment by pgonzal** [comment]
@srideshpande after this PR, we should be able to get your OUIFR packages enabled.

File: libraries/api-extractor/src/extractor/ApiExtractor.ts [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    path.join(__dirname, './api-extractor-config.schema.json'));
+
+  private _config: IExtractorConfig;
+  private _program: ts.Program;
+  private _localBuild: boolean;
+  private _logger: ILogger;
+  private _absoluteRootFolder: string;
+
+  private static _applyConfigDefaults(config: IExtractorConfig): IExtractorConfig {
+    const normalized: IExtractorConfig  = lodash.clone(config);
```
Comment (pgonzal) [inline libraries/api-extractor/src/extractor/ApiExtractor.ts]: >const [](start = 4, length = 5)

(QZ) Use lodash.merge()?

File: libraries/api-extractor/src/ExternalApiHelper.ts [hunk 2]
```diff
@@ -55,17 +47,31 @@ export default class ExternalApiHelper {
...
+      apiJsonFile: {
+        enabled: true,
+        outputFolder: outputPath
+      }
+    }, {
+       customLogger: {
+        logVerbose: (message: string) => { /* don't log */ },
+        logInfo: (message: string) => console.log(message),
+        logWarning: (message: string) => console.error(message),
+        logError: (message: string) => console.error(message)
```
Comment (iclanton) [inline libraries/api-extractor/src/ExternalApiHelper.ts]: merge these into the defaults so only one can be overridden.

File: libraries/api-extractor/src/Extractor.ts [hunk 3]
```diff
@@ -60,7 +44,6 @@ export interface IExtractorAnalyzeOptions {
  * @public
```
Comment (iclanton) [inline libraries/api-extractor/src/Extractor.ts]: >@public [](start = 3, length = 7)

remove the @public tag

**Comment by iclanton** [comment]
Approved.

**Pull request merged**